### PR TITLE
Add number of local devices to Room Details Admin API

### PR DESCRIPTION
**Pull request by dklimpel**
### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)
* [x] Code style is correct (run the [linters](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#code-style))

Signed-off-by: Dirk Klimpel dirk@klimpel.org

### CODE REVIEW

File: synapse/storage/databases/main/devices.py [hunk 1]
```diff
@@ -57,6 +57,37 @@ def __init__(self, database: DatabasePool, db_conn, hs):
...
+        def count_devices_by_users_txn(txn, user_ids):
+            sql = """
+                SELECT count(*)
+                FROM devices
+                WHERE
+                    hidden = '0' AND
+                    user_id IN ({})
+            """.format(
+                ",".join("?" for _ in user_ids)
+            )
```
Comment (erikjohnston) [inline synapse/storage/databases/main/devices.py:79]: There's a helper function called `make_in_list_sql_clause` that makes this easier, e,g.:

https://github.com/matrix-org/synapse/blob/df4b1e9c74d56d79c274149b0dfb0fd5305c7659/synapse/storage/databases/main/events_worker.py#L920-L924

**Review by erikjohnston — COMMENTED** [review]
Otherwise looks good!

File: docs/admin_api/rooms.md [hunk 2]
```diff
@@ -76,7 +76,7 @@ GET /_synapse/admin/v1/rooms
 
 Response:
 
-```
+```jsonc
```
Comment (clokep) [inline docs/admin_api/rooms.md:79]: What's `jsonc` vs. `json`? (It seems to work, just not seen this before.)

Comment (erikjohnston) [reply docs/admin_api/rooms.md:79]: I think its json with comments?

Comment (dklimpel) [reply docs/admin_api/rooms.md:79]: It is necessary for lines like `... (8 hidden items) ...`
Otherwise they will be marked as errors.

Comment (clokep) [reply docs/admin_api/rooms.md:79]: Great trick! Thanks!

**Review by erikjohnston — APPROVED** [review]
:tada:

**Pull request merged**